src/lib/core/WeaveCircularTLVBuffer.h

```diff
@@ -70,6 +70,8 @@ class NL_DLL_EXPORT WeaveCircularTLVBuffer
     inline size_t AvailableDataLength(void) const { return mQueueSize - mQueueLength; };
     inline size_t GetQueueSize(void) const { return mQueueSize; };
     inline uint8_t *GetQueue(void) const { return mQueue; };
+    inline void SetQueueHead(uint8_t *aQueueHead) { mQueueHead = aQueueHead; };
```

Comment:
These are not intended to be publicly settable.

Reply:
I guess this is probably a dumb question - then I should add CircularEventBuffer as a friend class to get access to private members of WeaveCircularTLVBuffer?

Reply:
No.  Because we can't have WeaveCircularTLVBuffer reference classes in the WDM profile.

I think the right way to do this is to make CircularEventBuffer a subclass of WeaveCircularTLVBuffer and then make these fields protected.

I'm okay if you want to leave the code as-is for now.